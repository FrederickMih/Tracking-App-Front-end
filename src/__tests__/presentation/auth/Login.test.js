import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureAppStore from '../../../Redux/configureStore';
import Login from '../../../components/presentation/auth/Login';

const store = configureAppStore();

describe('Login', () => {
  it('should match the snapshot of Login component', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Login />
          </BrowserRouter>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
