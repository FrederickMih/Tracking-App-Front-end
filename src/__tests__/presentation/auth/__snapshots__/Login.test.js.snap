// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot of Login component 1`] = `
<div
  className="auth"
>
  <div
    className="auth-sub"
  >
    <p>
      Are you a new user?
    </p>
    <a
      href="/registration"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
  <div
    className="error-message hide"
  >
    Sign In Failed: Invalid username or password
  </div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Username"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
